unikernel.ml

```diff
@@ -66,6 +101,24 @@ struct
       ~on_eof:(on_eof f_init);
     finished >>= fun data -> data
 
+  let extract_csrf_token reqd =
+    decode_request_body reqd >>= fun data ->
+    let json =
+      try Ok (Yojson.Basic.from_string data)
+      with Yojson.Json_error s -> Error (`Msg s)
+    in
+    match json with
```

Comment:
```suggestion
    match
      try Ok (Yojson.Basic.from_string data)
      with Yojson.Json_error s -> Error (`Msg s)
    with
```

You tend to introduce a lot of let-bindings, and I personally find fewer bindings make the code easier to understand, so I'd go and move the stuff into the `match .. with`. What do you think? (Note: this does not need to be applied here, we can as well go and refactor the codebase)

Reply:
oh, here actually in the bottom you return `json`, so don't mind the above.

Reply:
see also #69